logisland-core/logisland-api/src/main/java/com/hurence/logisland/record/Field.java

```diff
@@ -166,7 +166,26 @@ public Long asLong() {
 
     @Override
     public Boolean asBoolean() {
-        return (rawValue == null) ? null : (boolean) rawValue;
+        if(rawValue == null)
+            return null;
+
+        try{
+            boolean result = (boolean) rawValue;
+            return result;
+        }catch (Exception ex){
+
+
+                boolean result = asString().equalsIgnoreCase("y") ||
```

Comment:
asString may be called once (it is maybe better).

As well commons-lang BooleanUtils has a method to do it (if it's already in the dependencies it can be used): https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/BooleanUtils.html

(see toBoolean method)

Reply:
done. thanks for the advice